beacon-chain/core/blocks/block_operations.go

```diff
@@ -105,6 +72,43 @@ func verifyBlockRandao(proposer *pb.ValidatorRecord, block *pb.BeaconBlock) erro
 	return nil
 }
 
+// ProcessEth1Data is an operation performed on each
+// beacon block to ensure the ETH1 data votes are processed
+// into the beacon state.
+func ProcessEth1Data(beaconState *pb.BeaconState, block *pb.BeaconBlock) (*pb.BeaconState, error) {
+	if block.Eth1Data.DepositRootHash32 == nil {
```

Comment:
Why do we need to check for `nil` here?

1.) we are not doing it for any of the block operations
2.) if there are nils then block won't get accepted at first place, and the proposer who proposes it will lose $